rivertile/main.zig

```diff
@@ -14,129 +14,322 @@
+                    y += @intCast(i32, padding);
+                    width -= 2 * padding;
+                    height -= 2 * padding;
+
+                    switch (self.orientation) {
+                        .left => layout.pushViewDimensions(
+                            data.serial,
+                            x,
+                            y,
+                            width,
+                            height,
+                        ),
+                        .right => layout.pushViewDimensions(
+                            data.serial,
+                            @intCast(i32, usable_width - width) - x,
+                            y,
+                            width,
+                            height,
+                        ),
+                        .top => layout.pushViewDimensions(
+                            data.serial,
+                            y,
+                            x,
+                            height,
+                            width,
+                        ),
+                        .bottom => layout.pushViewDimensions(
+                            data.serial,
+                            y,
+                            @intCast(i32, usable_width - width) - x,
+                            height,
+                            width,
+                        ),
+                    }
+                }
+
+                // Layout pseudo-symbols are taken from dwm
+                layout.commit(data.serial, switch (self.orientation) {
+                    .left => "[]=",
+                    .right => "=[]",
+                    .top => "TTT",
+                    .bottom => "LLL",
+                });
+            },
+            .advertise_view => {},
+            .advertise_done => {},
         }
     }
+};
 
-    try stdout_buf.flush();
-}
+pub fn main() !void {
+    const display = wl.Display.connect(null) catch {
+        std.debug.warn("Unable to connect to Wayland server.\n", .{});
+        std.os.exit(1);
+    };
+    defer display.disconnect();
+
+    var context: Context = .{};
+
+    const args = std.os.argv;
+    if (args.len == 2) {
+        context.view_padding = try std.fmt.parseInt(u32, std.mem.spanZ(args[1]), 10);
+    }
 
-fn printUsageAndExit() noreturn {
-    const usage: []const u8 =
-        \\Usage: rivertile left|right|top|bottom [args passed by river]
-        \\
-    ;
+    const registry = try display.getRegistry();
+    try registry.setListener(*Context, registryListener, &context);
+    _ = try display.roundtrip();
 
-    std.debug.warn(usage, .{});
-    std.os.exit(1);
+    if (context.layout_manager == null) {
+        std.debug.warn("Wayland server does not support zriver_layout_unstable_v1.\n", .{});
+        std.os.exit(1);
+    }
+
+    context.configureAllOutputs();
+    defer context.destroyAllOutputs();
+
+    while (context.running) {
+        _ = display.dispatch() catch {
+            context.running = false;
+        };
```

Comment:
Let's not exit cleanly if there's an error.
```suggestion
        _ = try display.dispatch();
```

Reply:
That makes rivertile always error out when river closes, which I think is not good behaviour, considering this will be the one main way rivertile will be stopped.